app/abilities/ability.rb

```diff
@@ -54,6 +54,7 @@ def initialize(user)
         can [:index, :create, :destroy], UserMute
 
         if user.moderator?
+          can :show_redactions, [Node, Way, Relation, OldNode, OldWay, OldRelation]
```

Comment:
https://github.com/openstreetmap/openstreetmap-website/pull/4553/commits/012e752c0eae9450ff4ddf6083ad729779fd6eab

Reply:
Yes, I've seen this commit. I found the old code a bit clearer, because to me the concept of redactions makes only sense when talking about Old* elements.

Reply:
Checking `current_user&.moderator?` could be clearer only if you know that moderators can view redacted information.

Reply:
Tried replacing `@feature` with things like `"Old#{@type.classify}".constantize`, looks worse.

I don't see anything wrong with being able to "show_redactions" of "Node" (= "show_redactions" of some versions of "Node").

Reply:
"Node" (as in app/models/node.rb) does not have a redaction_id and does not belong to a redaction (unlike app/models/old_node.rb).

Also, "Node" has no concept of "some versions", it's always the latest version of an object only, which by definition cannot be redacted. Only the "Old*" object have the full element history, including the latest version.

So maybe I'm a bit confused here, because my definition of Node and OldNode is based on the model definitions.

Reply:
> "Node" has no concept of "some versions", it's always the latest version ...

Yet we link to its history from its page. How can we do that if "Node" is only the latest version?

Reply:
Well, to link to the history, you would only need the element type, object id and the current version of the object (=latest version). Based on this info, you could generate a link to the latest version, and also to the first version, if the current version is not 1.

The actual history page data would then be provided by the old* objects: https://github.com/openstreetmap/openstreetmap-website/blob/012e752c0eae9450ff4ddf6083ad729779fd6eab/app/views/browse/history.html.erb#L5

I found it helpful to check the local PostgreSQL db and compare current_nodes and nodes table contents (representing Node and OldNode models).